client/app/components/dashboards/dashboard-widget/VisualizationWidget.jsx

```diff
@@ -178,11 +180,17 @@ function VisualizationWidgetFooter({ widget, isPublic, onRefresh, onExpand }) {
         {!isPublic && (
           <a
             className="btn btn-sm btn-default hidden-print btn-transparent btn__refresh"
+            role="button"
+            tabIndex={0}
```

Comment:
Again, it doesn't feel like you should have to specify a `tabIndex` here -- is there something above it remove the `tabIndex` or something like that?

Reply:
Same story: anchors are considered lacking semantic meaning, therefore they require roles. If the role is interactive, a tabindex must be added, particularly if event handlers are present. I agree theoretically it is not necessary, but for my understanding it is sort of a safeguard.

Reply:
I'll defer to our [conversation above](https://github.com/getredash/redash/pull/5365#discussion_r568129610) so we can close out this thread. Rafael is correct that this needs a tabindex.